components/frontend/src/report/ReportsOverview.test.js

```diff
@@ -45,12 +61,48 @@ it('shows the comment', async () => {
     expect(screen.getAllByText(/Commentary/).length).toBe(1);
 });
 
+const reports = [
+    {
+        report_uuid: "report_uuid",
+        subjects: {
+            subject_uuid: {
+                metrics: {
+                    metric_uuid: {
+                        recent_measurements: [],
+                        tags: ["Foo"],
+                        type: "metric_type"
+                    },
+                    metric_uuid2: {
+                        recent_measurements: [],
+                        tags: ["Bar"],
+                        type: "metric_type"
+                    }
+                },
+                type: "subject_type",
+            }
+        }
+    }
+]
+
+const reportsOverview = { title: "Overview", permissions: {} }
+
+it('hides the report tag cards', async () => {
+    const toggleHiddenTag = jest.fn()
+    await act(async () => render_reports_overview(reports, reportsOverview, new Date(), toggleHiddenTag))
+    expect(screen.getAllByText(/Foo/).length).toBe(2)  // One in the dashboard, one in the table of metrics
+    expect(screen.getAllByText(/Bar/).length).toBe(2)  // One in the dashboard, one in the table of metrics
+    fireEvent.click(screen.getAllByText(/Foo/)[0])
+    expect(toggleHiddenTag).toHaveBeenLastCalledWith("Bar")
+})
+
 it('shows the report tag cards', async () => {
-    const reports = [{ subjects: { subject_uuid: { metrics: { metric_uuid: { tags: ["Tag"] } } } } }]
-    const reportsOverview = { title: "Overview", permissions: {} }
-    await act(async () => render_reports_overview(reports, reportsOverview))
-    expect(screen.getAllByText(/Tag/).length).toBe(1);
-});
+    const toggleHiddenTag = jest.fn()
+    await act(async () => render_reports_overview(reports, reportsOverview, new Date(), toggleHiddenTag, ["Bar"]))
+    expect(screen.getAllByText(/Foo/).length).toBe(2)  // One in the dashboard, one in the table of metrics
+    expect(screen.queryAllByText(/Bar/).length).toBe(0)
+    fireEvent.click(screen.getAllByText(/Foo/)[0])
+    expect(toggleHiddenTag).toHaveBeenLastCalledWith("Bar")
```

Comment:
Shouldn't we now check if the length of Bar objects is 2? Also checking for 0 at the end of the previous test might be overkill, although it would make the test-logic contained in the single case

Reply:
No, we can't, because the reports overview is not updated in this test after clicking. The event handler (toggleHiddenTag) is mocked.